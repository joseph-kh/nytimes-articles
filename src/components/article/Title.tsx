import React from "react";
import { TextProps } from "react-native";
import BoldText from "@/components/ui/texts/BoldText";

type Props = {
  title: string;
  fontSize?: number;
};

const Title: React.FC<Props & TextProps> = (props) => {
  const { title } = props;

  return <BoldText {...props}>{title}</BoldText>;
};

export default Title;
