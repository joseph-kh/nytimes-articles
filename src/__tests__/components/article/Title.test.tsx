import React from "react";
import renderer from "react-test-renderer";
import Title from "@/components/article/Title";

describe("Title Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Title title="Sample Title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
