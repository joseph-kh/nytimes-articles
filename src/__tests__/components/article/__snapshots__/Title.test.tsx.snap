// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders correctly 1`] = `
<Text
  style={
    {
      "fontFamily": "Bold",
      "fontSize": 21,
    }
  }
  title="Sample Title"
>
  Sample Title
</Text>
`;
